crates/zksync/core/src/cheatcodes.rs

```diff
@@ -94,7 +94,13 @@ where
     let zk_address = address.to_h160();
     let nonce_key = get_nonce_key(&zk_address).key().to_ru256();
     ecx.touch(&nonce_addr);
-    ecx.sstore(nonce_addr, nonce_key, tx_nonce.to_ru256()).expect("failed storing value");
+    // We make sure to keep the old deployment nonce
+    let old_deploy_nonce = ecx
+        .sload(nonce_addr, nonce_key)
+        .map(|v| decompose_full_nonce(v.to_u256()).1)
```

Comment:
Please add this functionality in `foundry_zksync_core` if not already present with well defined names and args, i.e. `get/set_nonces`, `get_deployment_nonce` `get_transaction_nonce`, etc. That helps us encapsulate zksync specific behaviour and that crate is already included everywhere.